Mathlib/GroupTheory/Perm/Cycle/Basic.lean

```diff
@@ -100,6 +100,12 @@ theorem SameCycle.trans : SameCycle f x y → SameCycle f y z → SameCycle f x
   fun ⟨i, hi⟩ ⟨j, hj⟩ => ⟨j + i, by rw [zpow_add, mul_apply, hi, hj]⟩
 #align equiv.perm.same_cycle.trans Equiv.Perm.SameCycle.trans
 
+theorem SameCycle.equivalence : Equivalence (SameCycle f) :=
+  ⟨SameCycle.refl f, SameCycle.symm, SameCycle.trans⟩
+
+def SameCycle.setoid (f : Perm α) : Setoid α :=
+  ⟨SameCycle f, SameCycle.equivalence⟩
```

Comment:
```suggestion
def SameCycle.setoid (f : Perm α) : Setoid α where
  iseqv := SameCycle.equivalence
```

Reply:
Thanks for the suggestion – I tried that locally and got:

```
don't know how to synthesize implicit argument
  @equivalence α (?m.33592 setoid f)
context:
ι: Type u_1
α: Type u_2
β: Type u_3
f✝g: Perm α
p: α → Prop
xyz: α
f: Perm α
⊢ Perm α
```

I guess Lean can't figure out the `f` 🤔

Reply:
`@SameCycle.equivalence _ f` does work. Though if we're going down that route, should we just make the `f` on `SameCycle.equivalence` an explicit parameter instead?